File: mode.rb
# http://rosettacode.org/wiki/Averages/Mode

test_array1 = [1, 2, 3, 1, 2, 4, 2, 5, 2, 3, 3, 1, 3, 6]
test_array2 = [1,1,1,1,1,2,2,3,3,3,3]

def find_mode(array)
  results = Hash.new
  array.each do |element|
    if !results.include?(element)
      results[element] = 1
    else
      results[element] += 1
    end
  end
  max_key = array[0]
  max_value = 1
  results.each do |key,val|
    if results[key] > max_value
      max_key = key
      max_value = val
    end
  end
  max_key
end

p find_mode(test_array1)
p find_mode(test_array2)

File: mode.swift
// http://rosettacode.org/wiki/Averages/Mode

func findMode(array:[Int]) -> Int {
  var hash = Dictionary<Int, Int>()
  for var i = 0; i < array.count; i++ {
    if hash[array[i]] == nil {
      hash[array[i]] = 1
    } else {
      hash[array[i]]! = hash[array[i]]! + 1
    }
  }
  var maxValue:Int = 1
  var maxKey:Int =  array[0]
  for (key,value) in hash {
    if value > maxValue {
      maxKey = key
      maxValue = value
    }
  }
  return maxKey
}

let values = [1, 2, 3, 1, 2, 4, 2, 5, 2, 3, 3, 1, 3, 6]
let values2 = [1,1,1,1,1,2,2,3,3,3,3]
println(findMode(values2))
